cameo/strain_design/deterministic/linear_programming.py

```diff
@@ -426,3 +430,394 @@ def _repr_html_(self):
+                heterologous_y_vars[y_var] = reaction
+            else:
+                raise RuntimeError("Something is wrong")
+
+        logger.debug("Control variables created")
+
+        # Set the objective
+        primal_objective = model.solver.objective
+        dual_objective = interface.Objective.clone(
+            dual_problem.objective, model=model.solver
+        )
+
+        reduced_expression = optlang.symbolics.Add(
+            *((c * v) for v, c in dual_objective.expression.as_coefficients_dict().items()
+              if v not in constrained_dual_vars)
+        )
+        dual_objective = interface.Objective(reduced_expression, direction=dual_objective.direction)
+
+        full_objective = interface.Objective(primal_objective.expression - dual_objective.expression, direction="max")
+        model.objective = full_objective
+        logger.debug("Objective created")
+
+        # Add number of knockouts constraint. ub=K+1 as the target will be forced to be 1 as well
+        knockout_number_constraint = model.solver.interface.Constraint(
+            optlang.symbolics.Add(*native_y_vars), lb=0, ub=n_knockouts + 1, name="number_of_knockouts_constraint"
+        )
+        model.solver.add(knockout_number_constraint)
+
+        # Add number of knockins constraint
+        knockin_number_constraint = model.solver.interface.Constraint(
+            optlang.symbolics.Add(*heterologous_y_vars), lb=0, ub=n_knockin, name="number_of_knockins_constraint"
+        )
+        model.solver.add(knockin_number_constraint)
+
+        # Add number of medium additions constraint
+        medium_additions_number_constraint = model.solver.interface.Constraint(
+            optlang.symbolics.Add(*medium_y_vars), lb=0, ub=n_medium, name="number_of_medium_additions_constraint"
+        )
+        model.solver.add(medium_additions_number_constraint)
+
+        logger.debug("Added constraint for number of knockouts, knockins and medium additions")
+
+        # Force target control variable to be 1, but allow flux in primal
+        model.solver.constraints["primal_y_const_" + target + "_lb"].lb = -2000
+        model.solver.constraints["primal_y_const_" + target + "_ub"].ub = 2000
+        model.variables["y_" + target].lb = 1
+
+        self.model = model
+        self.native_y_vars = native_y_vars
+        self.heterologous_y_vars = heterologous_y_vars
+        self.medium_y_vars = medium_y_vars
+
+        self.knockout_reactions = knockout_reactions
+        self.knockin_reactions = knockin_reactions
+        self.medium_additions = medium_additions
+
+        self.use_solution_pool = use_solution_pool
+
+        self.model.solver.configuration.presolve = True
+
+        if use_solution_pool:
+            if "gurobi_interface" not in model.solver.interface.__name__:
+                raise NotImplementedError("Solution pools are currently only available with the Gurobi solver")
+
+            self.model.solver.problem.params.PoolSearchMode = 2
```

Comment:
A comment here on what this means would be nice.
https://www.gurobi.com/documentation/8.1/refman/poolsearchmode.html